### PR TITLE
[WIP] Concurrent Mode!

### LINKED ISSUE #169: Concurrent Mode
[**Concurrent Mode**](https://reactjs.org/docs/concurrent-mode-intro.html) (with [**Suspense for Data Fetching**](https://reactjs.org/docs/concurrent-mode-suspense.html)) is now available in `react@experimental`. Creating this issue to track the work to update ReactFire to work with the latest and greatest.

### PR DESCRIPTION
**Pull request by jhuleatt**
Fixes #169 

Exploring Concurrent Mode

- [x] Rely on React experimental builds [so we get the new Concurrent Mode stuff](https://reactjs.org/docs/concurrent-mode-adoption.html#installation)
- [x] Support [preloading](https://reactjs.org/docs/concurrent-mode-suspense.html#for-library-authors)
- [x] Have a built-in way to lazy load the Firebase SDKs. Otherwise, it's really hard to create Firestore/RTDB queries for a preload
- [x] Update the sample to include the new [suggested patterns](https://reactjs.org/docs/concurrent-mode-patterns.html) so that we know ReactFire works well with them
- [x] Document the new/changed stuff

### CODE REVIEW

**Comment by jhuleatt** [comment]
Preview docs [here](https://github.com/FirebaseExtended/reactfire/tree/concurrent-mode/docs)!

File: docs/quickstart.md [hunk 1]
```diff
@@ -95,18 +90,19 @@ npm install firebase reactfire@canary
    ```jsx
    //...
    function Burrito() {
-     // create a ref
-     const firebaseApp = useFirebaseApp();
-     const burritoRef = firebaseApp
-       .firestore()
+     // lazy load the Firestore SDK
+     const firestore = useFirestore();
```
Comment (jhuleatt) [inline docs/quickstart.md]: @davideast @jamesdaniels How does this pattern look for lazy loading? It throws a Promise until the SDK is loaded

Comment (jamesdaniels) [reply docs/quickstart.md]: LGTM, do you want to provide `use*` options for the other components, even if we aren't utilizing them internally yet. e.g, `useFunctions()`, `useAnalytics()`

Comment (jhuleatt) [reply docs/quickstart.md]: yeah, I think that's a good call. will do

Comment (jhuleatt) [reply docs/quickstart.md]: done

**Pull request merged**